torch/_functorch/_aot_autograd/schemas.py

```diff
@@ -84,6 +110,15 @@ class OutputAliasInfo:
     dynamic_dims: Optional[Set[int]]
     # requires_grad
     requires_grad: bool
+    # FunctionalTensorWrapper that represents this output.
+    #
+    # Provides us the means to replay views from it.
+    #
+    # We need to wrap the actual FunctionalTensorWrapper with this class so that
+    # we only compare the tensor's metadata. That's because with the transformations
```

Comment:
Instead of the new class type, can we just use `has_same_metadata()`? https://github.com/pytorch/pytorch/blob/main/torch/_functorch/_aot_autograd/functional_utils.py#L251

It also does the right thing for dynamic shapes, and compares some other metadata like is_conj

Reply:
Although we can use `has_same_metadata()` function, I think the new class is still necessary. That's because [here](https://github.com/pytorch/pytorch/blob/ae6f8d923ceb032d3e933507b071284ba7ec218d/torch/_functorch/_aot_autograd/runtime_wrappers.py#L560-L568) and [here](https://github.com/pytorch/pytorch/blob/ae6f8d923ceb032d3e933507b071284ba7ec218d/torch/_functorch/_aot_autograd/runtime_wrappers.py#L726-L735), we call `OutputAliasInfo.__eq__`, which will check the equality for `functional_tensor`.

Reply:
Yep, that's fair. It's a bit unfortunate because those checks wouldn't exist our tracing were fast enough that we could just retrace when we hit each of those wrappers, but the extra wrapper class seems ok.